### PR TITLE
RTL1: Multicast demux support + cocotb tests enhancements

### LINKED ISSUE #1: RTL1 - Core Switching Primitives
This task focuses on assembling and extending the fundamental datapath components required for packet-switched communication within the NoC. Wherever applicable, existing, well-tested modules from the TAXI library will be reused as baseline building blocks (e.g. FIFOs, arbitration units, and flow control elements), reducing development effort and improving reliability.

Development efforts will concentrate on missing or NoC-specific primitives not covered by TAXI. This includes custom crossbar or switch matrix elements, width adaptation modules, and clock domain crossing (CDC) bridges tailored for Ethernet-style packet transport. Additional glue logic and interface wrappers will be introduced to ensure consistent integration of TAXI components into the openENOC architecture.

A key emphasis is placed on supporting heterogeneous interconnect conditions, including varying data path widths and differing clock domains. This involves the design of rate-matching buffers and interface adapters to enable seamless communication between components operating at different bus widths and clock frequencies.

The outcome of RTL1 is a cohesive, parameterizable set of switching primitives combining reused TAXI components with newly developed modules, forming the foundation of a flexible and interoperable switching fabric.

### PR DESCRIPTION
**Pull request by kerimbavcic**
- Created the structure
- Created a new openenoc_axis_demux module that supports multicast
- Created a new test infrastructure that tests both unicast and multicast with edge cases

### CODE REVIEW

File: dv/core/openenoc_axis_demux/Makefile [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# SPDX-FileCopyrightText: 2026 Kerim Bavcic
+# SPDX-License-Identifier: CERN-OHL-S-2.0
```
Comment (eniokaljic) [inline dv/core/openenoc_axis_demux/Makefile]: Change license to: AGPL-3.0-or-later

File: dv/core/openenoc_axis_demux/README.md
Comment (eniokaljic) [inline dv/core/openenoc_axis_demux/README.md:1]: Add license (CC-BY-SA-4.0) and copyright info

File: dv/core/openenoc_axis_demux/test_openenoc_axis_demux.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
+# SPDX-FileCopyrightText: 2026 Kerim Bavcic
+# SPDX-License-Identifier: CERN-OHL-S-2.0
```
Comment (eniokaljic) [inline dv/core/openenoc_axis_demux/test_openenoc_axis_demux.py]: Change license to: AGPL-3.0-or-later

File: dv/core/openenoc_axis_demux/test_openenoc_axis_demux.sv [hunk 3]
```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: 2026 Kerim Bavcic
+// SPDX-License-Identifier: CERN-OHL-S-2.0
```
Comment (eniokaljic) [inline dv/core/openenoc_axis_demux/test_openenoc_axis_demux.sv]: Change license to: AGPL-3.0-or-later

**Review by eniokaljic — CHANGES_REQUESTED** [review]

**Review by eniokaljic — APPROVED** [review]
